pkgs/by-name/te/tenv/package.nix

```
@@ -0,0 +1,37 @@
+{ buildGoModule, fetchFromGitHub, lib, testers, tenv }:
+
+buildGoModule rec {
+  pname = "tenv";
+  version = "1.2.0";
+
+  src = fetchFromGitHub {
+    owner = "tofuutils";
+    repo = "tenv";
+    rev = "v${version}";
+    hash = "sha256-yLNdBwKF6Jts661P+YZhFGNr71TG7Scb6RGvFxTLqYQ=";
+  };
+
+  vendorHash = "sha256-GGWiP1rIDF6qxST2ZmnKNkgbS+15hxaCs1d1+UEiYgU=";
+
+  # Tests disabled for requiring network access to release.hashicorp.com
+  doCheck = false;
+
+  passthru.tests.version = testers.testVersion {
+    package = tenv;
+    command = "HOME=$TMPDIR tenv --version";
+  };
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=v${version}"
+  ];
```

```suggestion
  ldflags = [
    "-s" "-w"
    "-X main.version=v${version}"
  ];

  passthru.tests.version = testers.testVersion {
    package = tenv;
    command = "HOME=$TMPDIR tenv --version";
  };
```

The derivation items are mostly ordered by 'phases order'.
`passthru` is always before meta.
ldflags is configure, etc. (Just for your reference.)